aws_lambda_powertools/utilities/idempotency/serialization/base.py

```diff
@@ -0,0 +1,19 @@
+"""
+Serialization for supporting idempotency
+"""
+from abc import ABC, abstractmethod
+from typing import Any, Dict
+
+
+class BaseIdempotencySerializer(ABC):
+    """
+    Abstract Base Class for Idempotency serialization layer, supporting dict operations.
+    """
+
+    @abstractmethod
+    def to_dict(self, data: Any) -> Dict:
+        pass
```

Comment:
Raising `NotImplementedError` is better because we increase the code coverage and follow the Python standards.

Reference: https://docs.python.org/3/library/exceptions.html#NotImplementedError

Reply:
Thanks!